### PR TITLE
Improve compatibility with flow

### PR DESCRIPTION
**Pull request by philippotto**
Nice package! :)

I tried to integrate flow (see [this](https://github.com/flowtype/flow-language-server) LSP library) with this ST package and found some issues with the hover support.

One issue was that the code didn't comply with the specification for the hover payload. The contents attribute can look like this `contents: MarkedString | MarkedString[];` (see [here](https://github.com/Microsoft/language-server-protocol/blob/master/protocol.md#textDocument_hover)), however it was implemented to only accept `contents: string | MarkedString[];`.

Another issue was that flow sometimes returns null instead of a MarkedString (array). This is flow's fault, but checking for that doesn't do any harm, I think.

The actual flow integration can be done by installing [this](https://github.com/flowtype/flow-language-server) package and adding the following to the config:

```
"flow":
      {
        "command": ["flow-language-server", "--stdio"],
        "scopes": ["source.js"],
        "syntaxes": ["Packages/Babel/JavaScript (Babel).sublime-syntax", "Packages/JavaScript/JavaScript.sublime-syntax"],
        "languageId": "javascript"
      },
```

### CODE REVIEW

**Comment by tomv564** [comment]
Thanks for reporting, this was totally a miss and your fix is elegant ! I'll do a quick review now.

File: main.py [hunk 1]
```diff
@@ -1466,6 +1466,9 @@ def on_hover(self, point, hover_zone):
 
     def handle_response(self, response, point):
         debug(response)
+        if response is None:
+            # Flow returns None sometimes
```
Comment (tomv564) [inline main.py]: Do you mind creating an issue for the Flow language server and linking it in the comment?

**Comment by philippotto** [comment]
I just found another problem. The spec says that the `triggerCharacters` attribute in `CompletionOptions` is optional. At the moment, LSP assumes that this is always defined.
However, I'm not sure how to resolve this best. If no `triggerCharacters` are defined, should we query for completions on every keystroke? Could this be a problem?

File: main.py (hunk 1 above)
Comment (philippotto) [inline main.py]: Sure, I'll do that!

**Comment by deathaxe** [comment]
We even should (see #25). I found top-level-completions such as keywords or imports (python) don't work as they are filtered. If ST queries for completion, LSP must forward it to the language server and show the results. How should LSP know whether the language server has something to offer or not?

`triggerCharacters` are special none-word characters, which would normally not trigger auto-completion. They are not a must have.

**Comment by tomv564** [comment]
Yes, let's try to figure out what the spec means with 'triggerCharacters' in #25. I'll merge this now, thanks for reporting the issue to the Flow server project!

**Comment by frou** [comment]
I'm curious about the potential of `flow`'s autocompletion engine.

Currently, [tern](http://ternjs.net) along with [tern_for_sublime](https://github.com/ternjs/tern_for_sublime) gives fantastically robust autocompletion for plain JS. 

I like the idea of flow and annotating JS (with [flow comments](https://flow.org/blog/2015/02/20/Flow-Comments/)) but until flow's autocompletion engine is as robust as tern's, it's too strange to also have to keep in mind an independent background type system that's different from what autocompletion is showing.

**Pull request merged**